okio/jvm/src/main/java/okio/Buffer.kt

```diff
@@ -0,0 +1,2176 @@
+      while (offset < size) {
+        val data = s!!.data
+        var pos = (s.pos + fromIndex - offset).toInt()
+        val limit = s.limit
+        while (pos < limit) {
+          val b = data[pos].toInt()
+          for (t in targetByteArray) {
+            if (b == t.toInt()) return pos - s.pos + offset
+          }
+          pos++
+        }
+
+        // Not in this segment. Try the next one.
+        offset += (s.limit - s.pos).toLong()
+        fromIndex = offset
+        s = s.next
+      }
+    }
+
+    return -1L
+  }
+
+  override fun rangeEquals(offset: Long, bytes: ByteString) =
+      rangeEquals(offset, bytes, 0, bytes.size())
+
+  override fun rangeEquals(
+    offset: Long, bytes: ByteString, bytesOffset: Int, byteCount: Int
+  ): Boolean {
+    if (offset < 0L
+        || bytesOffset < 0
+        || byteCount < 0
+        || size - offset < byteCount
+        || bytes.size() - bytesOffset < byteCount) {
+      return false
+    }
+    for (i in 0 until byteCount) {
+      if (getByte(offset + i) != bytes.getByte(bytesOffset + i)) {
+        return false
+      }
+    }
+    return true
+  }
+
+  /**
+   * Returns true if the range within this buffer starting at `segmentPos` in `segment` is equal to
+   * `bytes[bytesOffset..bytesLimit)`.
+   */
+  private fun rangeEquals(
+    segment: Segment, segmentPos: Int, bytes: ByteString, bytesOffset: Int, bytesLimit: Int
+  ): Boolean {
+    var segment = segment
+    var segmentPos = segmentPos
+    var segmentLimit = segment.limit
+    var data = segment.data
+
+    var i = bytesOffset
+    while (i < bytesLimit) {
+      if (segmentPos == segmentLimit) {
+        segment = segment.next!!
+        data = segment.data
+        segmentPos = segment.pos
+        segmentLimit = segment.limit
+      }
+
+      if (data[segmentPos] != bytes.getByte(i)) {
+        return false
+      }
+
+      segmentPos++
+      i++
+    }
+
+    return true
+  }
+
+  override fun flush() {}
+
+  override fun isOpen() = true
+
+  override fun close() {}
+
+  override fun timeout() = Timeout.NONE
+
+  /** For testing. This returns the sizes of the segments in this buffer.  */
+  internal fun segmentSizes(): List<Int> {
+    if (head == null) return emptyList()
+    val result = ArrayList<Int>()
+    result.add(head!!.limit - head!!.pos)
+    var s = head!!.next
+    while (s != head) {
+      result.add(s!!.limit - s.pos)
+      s = s.next
+    }
+    return result
+  }
+
+  /** Returns the 128-bit MD5 hash of this buffer.  */
+  @Suppress("NON_FINAL_MEMBER_IN_FINAL_CLASS") // Required to keep JApicmp happy.
+  open fun md5() = digest("MD5")
+
+  /** Returns the 160-bit SHA-1 hash of this buffer.  */
+  @Suppress("NON_FINAL_MEMBER_IN_FINAL_CLASS") // Required to keep JApicmp happy.
+  open fun sha1() = digest("SHA-1")
+
+  /** Returns the 256-bit SHA-256 hash of this buffer.  */
+  @Suppress("NON_FINAL_MEMBER_IN_FINAL_CLASS") // Required to keep JApicmp happy.
```

Comment:
I’m tempted to cut an intermediate 1.x release that makes all of these final, just so we have that as our baseline

Reply:
Alternatively, once we're ready for 2.0 we disable japicmp, fix all these, release, and make 2.0 the baseline.

Reply:
Just a fly on the wall here, but @JakeWharton 's suggestion sounds reasonable to this observer.

Reply:
I ran into a wall trying to get the `size()` method on Options to work. I think we quietly release Okio 1.14.1 which just adds a bunch of redundant final modifiers.

https://github.com/square/okio/pull/417/files